files/image_config/reboot-cause/determine-reboot-cause

```diff
@@ -1,28 +1,33 @@
 #!/usr/bin/env python
 #
-# process-reboot-cause
+# determine-reboot-cause
 #
 # Program designed to run once, soon after system boot which will
 # determine the cause of the previous reboot and store it to the disk,
 #
 
 try:
+    import datetime
+    import json
     import os
     import pwd
     import re
     import sys
 
+    import swsssdk
     from sonic_py_common import device_info, logger
+
 except ImportError as err:
     raise ImportError("%s - required module not found" % str(err))
 
 VERSION = "1.0"
 
-SYSLOG_IDENTIFIER = "process-reboot-cause"
+SYSLOG_IDENTIFIER = "determine-reboot-cause"
 
 REBOOT_CAUSE_DIR = "/host/reboot-cause/"
+PREVIOUS_REBOOT_CAUSE_DIR = "/host/reboot-cause/previous-reboot-cause/"
 REBOOT_CAUSE_FILE = REBOOT_CAUSE_DIR + "reboot-cause.txt"
-PREVIOUS_REBOOT_CAUSE_FILE = REBOOT_CAUSE_DIR + "previous-reboot-cause.txt"
+PREVIOUS_REBOOT_CAUSE_FILE = REBOOT_CAUSE_DIR + "last-reboot-cause.json"
```

Comment:
Prefer sticking with "previous-" rather than "last-", as it's a bit more specific, whereas "last" can have a few meanings.